### ISSUE TITLE
Make extensible enums actually extensible

### ISSUE THREAD

**Issue by atollk** [cjbooms/fabrikt#213]
Right now, using `x-extensible-enum` to declare an enum instead of the `enum` keyword does not actually generate "extensible enums", as the name suggests. Instead, it's just an alias for "enum". For example, this enum model here:

```kotlin
enum class ExtensibleEnumObject(
    @JsonValue
    val value: String
) {
    ACTIVE("active"),

    INACTIVE("inactive");

    companion object {
        private val mapping: Map<String, ExtensibleEnumObject> =
            values().associateBy(ExtensibleEnumObject::value)

        fun fromValue(value: String): ExtensibleEnumObject? = mapping[value]
    }
}
```

should actually look more like this when marked as "extensible": (untested code)

```kotlin
enum class ExtensibleEnumObject(
    @JsonValue
    val value: String
) {
    ACTIVE("active"),

    INACTIVE("inactive"),

    UNKNOWN("");

    companion object {
        private val mapping: Map<String, ExtensibleEnumObject> =
            values().filter { it != UNKNOWN }.associateBy(ExtensibleEnumObject::value)

        @JsonCreator
        fun fromValue(value: String): ExtensibleEnumObject = mapping[value] ?: UNKNOWN
    }
}
```

**Comment by cjbooms** [comment]
Interesting idea. The way we use this feature is we opt-in to the alias when generating the server-side but leave it disabled for client side. That way we get the strict typing server side, and our clients don't break when we extend with new values.

Your proposal might be a nice improvement though. 

Side note, I've a relatively clear calendar tomorrow and will try and catch up fabrik issues and PRs

**Comment by atollk** [comment]
> Interesting idea. The way we use this feature is we opt-in to the alias when generating the server-side but leave it disabled for client side. That way we get the strict typing server side, and our clients don't break when we extend with new values.

So you're mainly using it for extensibility in responses (i.e. the client might have to deal with unknown values) and my proposed change would offer extensibility in requests (i.e. the server might have to deal with unknown values). Correct?

> Side note, I've a relatively clear calendar tomorrow and will try and catch up fabrikt issues and PRs

Nice, thank you :)

**Comment by cjbooms** [comment]
Yes, at Zalando x-extensible-enum is used to communicate to clients that: these are the current options, but new options may be added in the future

To my knowledge, it is never used to allow clients to make requests using undefined enum values, only to prepare clients to gracefully handle responses with unexpected new enum values as the API evolves over time.

**Comment by cjbooms** [comment]
Added support for fault tolerant enums here 
https://github.com/fabrikt-io/fabrikt/pull/526

Closing this issue
